README.md

```diff
@@ -72,43 +79,69 @@ Now write the tools for exact pattern matching. You can use the naive algorithm
 
 ### Problems encountered if any. 
 
-*Describe this here.*
+ANSWER:
+I had some trouble jumping backwards (or rather figuring out that i had to repetitively jump back until i reached match) in my border-array (kind of a brain-fart i know) (finally fixed it with a while-loop). 
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+ANSWER:
+I simulate a bunch of random DNA sequences of varying lengths (50-100 nucleotides; representing ref seqs), and from these sequence sampled random intervals of length 1-20 representing exact-read seqs. These samples were then used to test if my implentations returned the exact intervals/positions (compared to re.findall(), see first section of src/runtime_comparison.py) through 500000 iterations. 
+Hereafter i similarily sampled ref/reads while feeding these to both algorithms (naive vs lin) simultaneously. The algorithms returned the same result for all iterations (500000; see src/runtime_comparison.py). A coverage test was performed using coverage.py (https://coverage.readthedocs.io/en/6.4.4/). For this purpuse a fasta file containing 5 sequences and a corresponding simple fastq file containing 100 reads were generated (using scripts provided by teacher in previus week, see data_for_testing/simulate_simple_fasta.py; data_for_testing/simulate_simple_fastq.py). The naive and lin algorithms had respectivly 95% and 94% coverage where all missing lines (naive: 15, 17, 21, 23; lin: 16, 32, 42, 44, 48, 50) was involved in early returnings as response to empty/incompatable inputs. These responses were then tested manually i.e. both algorithms were tested for varius empty input cases (empty/empty and non-empty/empty etc.) Sequences containing first and last 10 nucleotides repeatedly was used to test 'takeoff and landing'. 
+
 
 ### Experiments validating the running time.
 
 For this section, you should address the following:
 
+ANSWER:
+
+Runtimes for varying lengths of m and n for the naive algorithm:
+![](figs/Naive_runtimes.png)
+
+Runtimes for varying lengths of m and n for the linear algorithm:
+![](figs/Linear_runtimes.png)
+
+
 * An experiment that verifies that your implementation of `naive` uses no more time than O(nm) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs? 
 
+ANSWER:
+Worst case for the naive is when n and m are all repetitions of the same character e.g. n = 'AAAAAAAAAAAAA' m = 'AAA'. In this case the algorithm will have to compare all characters from both strings against each other (runtime = mn). Oppositly if n and m are all different e.g. n = 'AAAAAAAAAAAAA' m = 'TTT' the algorithm will only have to compare the first character of m against all n (runtime = n). 
+
+Runtimes naive best-case:
+![](figs/Naive_best_case.png)
+
+Runtimes naive worst-case:
+
+![](figs/Naive_worst_case.png)
+
```

Comment:
Looks good, but how do you determine best/worst case here? You only have an asymptotic complexity, so how do you plot that?